### DISCUSSION TITLE
Command palette :search

### DISCUSSION THREAD

**Discussion by UliTroyo** [helix-editor/helix#4792] [Ideas] [answered]
Hi! I was looking through the [palette commands](https://github.com/helix-editor/helix/blob/c6b83368b3b626cb63120b5ac684cc8b1b693172/helix-term/src/commands/typed.rs) to see if maybe there's a feature that would allow me to add a remap that would allow me to search for a specific pattern, but there doesn't seem to be a way. I know I can remap `search` and `search_selection`, which get me pretty far, but what I'd like is to remap `:search $$$`, which is how I mark generated code to jump around the bits that are custom. I get a lot of mileage out of having both a static `shell_pipe` and a `:pipe` palette command I can remap, so I was wondering if the same could be done to search?

**Comment by the-mikedavis** [comment, marked as answer]
This would be possible with macro keybinds (#1383, #4709)

```toml
[keys.normal]
M = "@/\\$\\$\\$<ret>"
```
